events/message.go

```diff
@@ -16,12 +16,79 @@
+	// ErrMissingChangeMessageSubjectID is returned when the event message has the incorrect field SubjectID value.
+	ErrMissingChangeMessageSubjectID = errors.New("change message SubjectID field required")
+
+	// ErrMissingEventMessageEventType is returned when the event message has the incorrect field EventType value.
+	ErrMissingEventMessageEventType = errors.New("event message EventType field required")
+	// ErrMissingEventMessageSubjectID is returned when the event message has the incorrect field SubjectID value.
+	ErrMissingEventMessageSubjectID = errors.New("event message SubjectID field required")
+
+	// ErrInvalidAuthRelationshipRequestAction is returned when the event message has the incorrect field Action value.
+	ErrInvalidAuthRelationshipRequestAction = errors.New("auth relationship request message Action field must be write or delete")
+	// ErrMissingAuthRelationshipRequestObjectID is returned when the event message has the incorrect field ObjectID value.
+	ErrMissingAuthRelationshipRequestObjectID = errors.New("auth relationship request message ObjectID field required")
+	// ErrMissingAuthRelationshipRequestRelation is returned when the event message has no relations defined.
+	ErrMissingAuthRelationshipRequestRelation = errors.New("auth relationship request message Relations field required")
+	// ErrMissingAuthRelationshipRequestRelationRelation is returned when the event message Relations has the incorrect field for Relation value.
+	ErrMissingAuthRelationshipRequestRelationRelation = errors.New("auth relationship request message Relations Relation field required")
+	// ErrMissingAuthRelationshipRequestRelationSubjectID is returned when the event message Relations has the incorrect field SubjectID value.
+	ErrMissingAuthRelationshipRequestRelationSubjectID = errors.New("auth relationship request message Relations SubjectID field required")
+)
+
+// Message contains a message which has been published or received from a subscription.
+type Message[T any] interface {
+	// Connection returns the underlying connection the message was received on.
+	Connection() Connection
+
+	// ID returns the unique message id.
+	ID() string
+	// Topic returns the topic the message was sent to.
+	Topic() string
+	// Message returns the decoded message object.
+	Message() T
+	// Ack acks the message.
+	Ack() error
+	// Nak nacks the message.
+	Nak(delay time.Duration) error
+	// Term terminates the message.
+	Term() error
+	// Timestamp returns the time the message was submitted.
+	Timestamp() time.Time
+	// Deliveries returns the number of times the message was delivered.
+	Deliveries() uint64
+
+	// Error returns any error encountered while decoding the message
+	Error() error
+
+	// Source returns the underlying message object.
+	Source() any
+}
+
+// Request extends Message by allowing replies to be sent for the received message.
+type Request[T, U any] interface {
```

Comment:
By just looking at this interface it's not clear to me what the types `T` and `U` represent.

```suggestion
type Replier[Request, Response any] interface {
```

I think this is clearer in intention because I read this as "The replier interface takes two type parameters, the request and response type"